ghcide/src/Development/IDE/Core/RuleTypes.hs

```diff
@@ -407,9 +408,15 @@ data GhcSession = GhcSession
 instance Hashable GhcSession
 instance NFData   GhcSession
 
-data GhcSessionDeps = GhcSessionDeps deriving (Eq, Show, Typeable, Generic)
-instance Hashable GhcSessionDeps
-instance NFData   GhcSessionDeps
+newtype GhcSessionDeps = GhcSessionDeps_
+    { -- | Load full ModSummary values in the GHC session.
+        -- Required for interactive evaluation, but leads to more cache invalidations
+        fullModSummary :: Bool
```

Comment:
Why do we put some things here and some things in `GhcSessionDepsConfig`? Should this field just be a `GhcSessionDepsConfig`, with a slightly-more-complex function for checking newness?

Reply:
`fullModSummary` is part of the build key because it has an effect on the build result.
`checkForImportCycles` is part of the config because it only effects whether an exception is thrown on module cycles (and how expensive the build is), but not the build result